src/Aspire.Hosting/Dcp/Process/ProcessUtil.cs

```diff
@@ -39,7 +39,9 @@ public static (Task<ProcessResult>, IAsyncDisposable) Run(ProcessSpec processSpe
             process.StartInfo.Environment[key] = value;
         }
 
-        var processEventLock = new object();
+        Use a reset event to prevent output processing and exited events from running until OnStart is complete.
+        OnStart might have logic that sets up data structures that then are used by these events.
```

Comment:
```suggestion
        // Use a reset event to prevent output processing and exited events from running until OnStart is complete.
        // OnStart might have logic that sets up data structures that then are used by these events.
```